dpctl/_sycl_context.pyx

```diff
@@ -140,6 +148,28 @@ cdef class SyclContext(_SyclContext):
         SyclContext._init_helper(<_SyclContext> self, CRef)
         return 0
 
+    cdef int _init_context_from_capsule(self, object cap):
+        """
+        For named PyCapsule with name SyclContextRef, which carries pointer to
+        sycl::context object, interpreted as DPCTLSyclContextRef, creates corresponding
+        SyclContext.
+        """
+        cdef DPCTLSyclContextRef CRef = NULL
+        cdef DPCTLSyclContextRef CRef_copy = NULL
+        cdef int ret = 0
+        if pycapsule.PyCapsule_IsValid(cap, "SyclContextRef"):
+            CRef = <DPCTLSyclContextRef> pycapsule.PyCapsule_GetPointer(cap, "SyclContextRef")
+            if (CRef is NULL):
+                return -6
+            ret = pycapsule.PyCapsule_SetName(cap, "used_SyclContextRef")
+            if (ret):
```

Comment:
what are you ?checking here

Reply:
`PyCapsule_SetName` return 0 on success and [non-zero on failure](https://docs.python.org/3/c-api/capsule.html#c.PyCapsule_SetName)